saleor/graphql/checkout/tests/mutations/test_checkout_lines_update.py

```diff
@@ -1809,3 +1809,109 @@ def test_checkout_lines_update_with_invalid_metadata(
 
     assert expected_error["field"] == "metadata"
     assert expected_error["code"] == "REQUIRED"
+
+
+def test_checkout_lines_update_with_empy_metadata_preserve_old(
+    user_api_client,
+    checkout_with_item,
+):
+    # given
+    checkout = checkout_with_item
+
+    assert checkout.lines.count() == 1
+
+    line = checkout.lines.first()
+
+    line.metadata["test_key"] = "old_value"
+    line.save(update_fields=["metadata"])
+
+    variant = line.variant
+
+    variant_id = graphene.Node.to_global_id("ProductVariant", variant.pk)
+
+    # when
+    variables = {
+        "id": to_global_id_or_none(checkout_with_item),
+        "lines": [
+            {
+                "variantId": variant_id,
+                "quantity": 1,
+                # Leave input empty to ensure it will not affect existing entries
+                "metadata": [],
+            }
+        ],
+    }
+    response = user_api_client.post_graphql(MUTATION_CHECKOUT_LINES_UPDATE, variables)
+
+    # then
+    content = get_graphql_content(response)
+
+    data = content["data"]["checkoutLinesUpdate"]
+
+    assert not data["errors"]
+
+    checkout.refresh_from_db()
+
+    assert checkout.lines.count() == 1
+
+    line = checkout.lines.first()
+
+    assert line.metadata["test_key"] == "old_value"
+
+    assert line.variant == variant
+
+
+def test_checkout_lines_update_with_empy_metadata_merge_old(
+    user_api_client,
+    checkout_with_item,
+):
+    # given
+    checkout = checkout_with_item
+
+    assert checkout.lines.count() == 1
+
+    line = checkout.lines.first()
+
+    line.metadata["test_key"] = "old_value"
+    line.save(update_fields=["metadata"])
+
+    variant = line.variant
+
+    variant_id = graphene.Node.to_global_id("ProductVariant", variant.pk)
+
+    # when
+    variables = {
+        "id": to_global_id_or_none(checkout_with_item),
+        "lines": [
+            {
+                "variantId": variant_id,
+                "quantity": 1,
+                # Leave input empty to ensure it will not affect existing entries
```

Comment:
Test name, and this comment says that passed metadata should be empty.

Reply:
invalid test name, it should merge old and new

previous test in this file is merging empty. updated